test/apps/iodemo/io_demo.cc

```diff
@@ -172,23 +224,30 @@ class P2pDemoCommon : public UcxContext {
     size_t                   _cur_buffer_idx;
     size_t                   _padding;
 };
+MemoryPool<P2pDemoCommon::IoMessage>  P2pDemoCommon::ioMsgPool;
 
 
 class DemoServer : public P2pDemoCommon {
 public:
     // sends an IO response when done
     class IoWriteResponseCallback : public UcxCallback {
     public:
-        IoWriteResponseCallback(DemoServer *server, UcxConnection* conn,
-                                uint32_t sn, size_t data_size) :
-            _server(server), _conn(conn), _sn(sn), _data_size(data_size) {
+        IoWriteResponseCallback() :
+            _server(NULL), _conn(NULL), _sn(0), _data_size(0) {
+        }
+
+        void init(DemoServer *server, UcxConnection* conn, uint32_t sn, size_t data_size) {
```

Comment:
why need separate init() function ?

Reply:
We need to use init() function to reset the values(op/sn/data_size) when we want to reuse the item was used and freed.